File: com/google/code/juds/test/TestUnixDomainSocket.py
#!/usr/bin/env python

import sys, os, socket, time

BUFSIZE = 128

def main():
    if len(sys.argv) != 2:
        print "usage: %s socketfilename" % sys.argv[0]
        sys.exit(1)
    socket_file = sys.argv[1]

    # Testcase 1.1: Test UnixDomainSocketClient with a stream socket
    print ("Test #1: Test UnixDomainSocketClient\n"
        "Testcase 1.1: Test UnixDomainSocketClient with a stream socket...")
    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    s.bind(socket_file)
    s.listen(0)
    client, client_address = s.accept()
    text = "[1] Hello I'm the server!"
    client.send(text)
    print "Text sent: \"%s\"" % text
    text = client.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()
    os.unlink(socket_file)
    
    # Testcase 1.2: Test UnixDomainSocketClient with a datagram socket
    print "Testcase 1.2: Test UnixDomainSocketClient with a datagram socket..."
    s = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
    s.bind(socket_file)
    text = s.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()
    os.unlink(socket_file)
    
    time.sleep(1)   # wait for the server

    # Testcase 2.1: Test UnixDomainSocketServer with a stream socket
    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    s.connect(socket_file)
    text = "[4] Hello I'm the client!"
    s.send(text)
    print "Text sent: \"%s\"" % text
    text = s.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()
    
    time.sleep(1)   # wait for the server

    # Testcase 2.2: Test UnixDomainSocketServer with a datagram socket
    s = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
    s.connect(socket_file)
    text = "[6] Hello I'm the client!"
    s.send(text)
    print "Text sent: \"%s\"" % text
    s.close()


if __name__ == "__main__":
    main()

File: test/TestUnixDomainSocket.java
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import com.etsy.net.*;

public class TestUnixDomainSocket {

	public static void main(String[] args) throws IOException, InterruptedException {
		if (args.length != 1) {
			System.out
					.println("usage: java TestUnixDomainSocket socketfilename");
			System.exit(1);
		}
		String socketFile = args[0];

		byte[] b = new byte[128];
		// Testcase 1.1: Test UnixDomainSocketClient with a stream socket
		UnixDomainSocketClient socket = new UnixDomainSocketClient(socketFile,
				JUDS.SOCK_STREAM);
		InputStream in = socket.getInputStream();
		OutputStream out = socket.getOutputStream();
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		String text = "[2] Hello! I'm the client!";
		out.write(text.getBytes());
		System.out.println("Text sent: " + "\"" + text + "\"");
		socket.close();

        Thread.sleep(2000);

		// Testcase 1.2: Test UnixDomainSocketClient with a datagram socket
		socket = new UnixDomainSocketClient(socketFile,
				JUDS.SOCK_DGRAM);
		System.out.println("Provoke and catch an "
				+ "UnsupportedOperationException:");
		try {
			in = socket.getInputStream();
		} catch (UnsupportedOperationException e) {
			System.out.println("UnsupportedOperationException has been "
					+ "thrown as expected.");
		}
		out = socket.getOutputStream();
		text = "[3] Hello! I'm the client!";
		out.write(text.getBytes());
		System.out.println("Text sent: \"" + text + "\"");
		socket.close();

        Thread.sleep(2000);

		// Testcase 1.3: Test UnixDomainSocketClient with a SEQPACKET socket
		socket = new UnixDomainSocketClient(socketFile,JUDS.SOCK_SEQPACKET);
		System.out.println("Provoke and catch an "
				+ "UnsupportedOperationException:");
		try {
			in = socket.getInputStream();
		} catch (UnsupportedOperationException e) {
			System.out.println("UnsupportedOperationException has been "
					+ "thrown as expected.");
		}
		out = socket.getOutputStream();
		text = "[3] Hello! I'm the client!";
		out.write(text.getBytes());
		System.out.println("Text sent: \"" + text + "\"");
		socket.close();

		// Testcase 2.1: Test UnixDomainSocketServer with a stream socket
		System.out.println("\nTest #2: Test UnixDomainSocketServer\nTestcase "
				+ "2.1: Test UnixDomainSocketServer with a stream socket...");
		UnixDomainSocketServer ssocket = new UnixDomainSocketServer(socketFile,
				JUDS.SOCK_STREAM);
		in = ssocket.getInputStream();
		out = ssocket.getOutputStream();
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		text = "[5] Hello! I'm the server!";
		out.write(text.getBytes());
		System.out.println("Text sent: " + "\"" + text + "\"");
		ssocket.close();
		ssocket.unlink();

		// Testcase 2.2: Test UnixDomainSocketServer with a datagram socket
		System.out.println("Testcase 2.2: Test UnixDomainSocketServer with "
				+ "a datagram socket...");
		ssocket = new UnixDomainSocketServer(socketFile,
				JUDS.SOCK_DGRAM);
		System.out.println("Provoke and catch an "
				+ "UnsupportedOperationException:");
		in = ssocket.getInputStream();
		try {
			out = ssocket.getOutputStream();
		} catch (UnsupportedOperationException e) {
			System.out.println("UnsupportedOperationException has been "
					+ "thrown as expected.");
		}
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		ssocket.close();
		ssocket.unlink();
	}
}
